opesci/staggeredgrid.py

```diff
@@ -1030,6 +1032,102 @@ def initialise_bc(self):
         result += self.velocity_bc.replace(t1, rep)
         return result
 
+    def simple_kernel(self, grid_field, indexes, template):
+        """
+        generate the inner loop with all fields from stress or velocity
+	:param grid_field: stress or velocity field array
+	:param indexes: array with dimension, dimension var, initial margin, final margin
+	:param template: mako string template
+        - iterate through fields and replace mako template
+        - return inner loop code as string
+        """
+        body = ''
+        idx = [self.time[1]] + self.index
+        for field in grid_field:
+           kernel = self.transform_kernel(field)
+           if self.read:
+                kernel = self.resolve_media_params(kernel)
+           body += ccode(field[idx]) + '=' \
+	            + ccode(kernel.xreplace({self.t+1: self.time[1], self.t: self.time[0]})) + ';\n'
+        dict1 = {'i': indexes[1], 'i0': indexes[2], 'i1': indexes[3], 'body': body}
+        body = render(template, dict1)
+        if not self.pluto and self.ivdep and indexes[0] == self.dimension-1:
+    	    body = '%s\n' % self.compiler._ivdep + body
+        if not self.pluto and self.simd and indexes[0] == self.dimension-1:
+    	    body = '#pragma simd\n' + body
+
+        return body
+
+    def fission_kernel(self, grid_field, indexes, template):
+        """
+        generate the inner loop with all fields from stress or velocity
+	:param grid_field: stress or velocity field array
+	:param indexes: array with dimension, dimension var, initial margin, final margin
+	:param template: mako string template
+        - iterate through fields and for each dimension separate minus, plus and unitary strides
+	  on its own loop replacing it on mako template
+        - return inner loop code as string
+	"""
+	body = ''
+        body_tmp = ''
+	sinal = ['=','+=']
```

Comment:
I've no idea what's going on in these two new functions so some comments would be nice. For example, what does `sinal` mean? Google Translate tells me that "sinal" is Portuguese for "sign" in English, which I assume is what this is referring to with the `=` and `+=`, but maybe `operator` would be a better variable name?